test/TestNamespaceTracker.hs

```diff
@@ -231,11 +231,11 @@ longHand i chan = reverse <$> inner []
 
 untilDisconnect ::
     (Eq i, Monad m) => i -> Protocol a a (ServerEvent i b) (ServerEvent i b) m ()
-untilDisconnect i = trace "until dis" $ waitThen sendFwd next
+untilDisconnect i = waitThen fwd next
   where
-    next e@(ServerDisconnect i')
-      | i' == i = sendRev e
```

Comment:
This was checking that a particular user was sending the disconnect (`i' == i`). Is that not what you want to do any more? If so, why are you still taking the `i`?

Reply:
I undo this later, I thought you could put the value in the pattern match, but it doesn't work like that.

Reply:
Ah, no, pattern matches just have variables as binding sites. I got bitten by that too.